dotnet-desktop-guide/net/winforms/visualstudio/troubleshoot-32bit.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Troubleshoot 32-bit problems
+description: Describes the problems with 32-bit components with Visual Studio 2022 and Windows Forms, and those problems can be fixed. Instructions on how to enable the out-of-process designer are provided.
+ms.date: 03/08/2024
+ms.topic: overview
+---
+
+# Troubleshoot 32-bit problems (Windows Forms .NET)
+
+After upgrading to Visual Studio 2022, you might run into a problem where the design-time experience of your app stops working. This could be related to referencing a 32-bit component. Visual Studio 2022 is a 64-bit process, and can't reference 32-bit components, regardless of the underlaying technology, such as .NET Framework, .NET, or COM\ActiveX. You might not realize you have references to 32-bit components until you try to upgrade Visual Studio. References that compile to 64-bit or target `AnyCPU` continue to work. You'll also run into the same problem if a component you're referencing compiles to `AnyCPU` but happens to reference something 32-bit.
+
+## What's the problem
+
+Windows Forms code runs in two modes: design-time and run-time. At run-time, you're running in whatever mode you compiled for: 32-bit or 64-bit; .NET Framework or .NET. At design-time, your code is run inside Visual Studio, which is a 64-bit .NET Framework process. If your project code doesn't match that environment, it can't run in the designer. For example, if your project is targeting 32-bit .NET Framework or 64-bit .NET, it doesn't match Visual Studio's 64-bit .NET Framework process. And that's the problem, the Windows Forms designer in Visual Studio can't instantiate 32-bit components or .NET components directly, it can only instantiate 64-bit .NET Framework components. To fix these integration issues, the Windows Forms team created the **out-of-process designer** for Visual Studio that acts like a translation layer for the Windows Forms designer. The out-of-process designer communicates with the Visual Studio 64-bit .NET Framework designer on behalf of your code so that you can use the designer with .NET projects.
```

Comment:
Changing a comma to a colon.
```suggestion
Windows Forms code runs in two modes: design-time and run-time. At run-time, you're running in whatever mode you compiled for: 32-bit or 64-bit; .NET Framework or .NET. At design-time, your code is run inside Visual Studio, which is a 64-bit .NET Framework process. If your project code doesn't match that environment, it can't run in the designer. For example, if your project is targeting 32-bit .NET Framework or 64-bit .NET, it doesn't match Visual Studio's 64-bit .NET Framework process. And that's the problem: the Windows Forms designer in Visual Studio can't instantiate 32-bit components or .NET components directly, it can only instantiate 64-bit .NET Framework components. To fix these integration issues, the Windows Forms team created the **out-of-process designer** for Visual Studio that acts like a translation layer for the Windows Forms designer. The out-of-process designer communicates with the Visual Studio 64-bit .NET Framework designer on behalf of your code so that you can use the designer with .NET projects.
```